### DISCUSSION TITLE
FAQ: How to create image in every single style with just one click

### DISCUSSION THREAD

**Discussion by docppp** [lllyasviel/Fooocus#1751] [Show and tell]
## Note: This instruction is intended for non-programmers

There has been a lot of discussion lately about a "style preview" to make it easier for the user to see what a particular style looks like. Since there's no one good way to integrate such a demo, here's a quick guide to generating a ton of images yourself. Keep in mind that python is space-sensitive so all lines must have proper indentation.

## 1. Disable auto-update on run.bat

Replace whole content of `entry_with_update.py` with following:
```
import os
import sys


root = os.path.dirname(os.path.abspath(__file__))
sys.path.append(root)
os.chdir(root)

print('Update succeeded.')
from launch import *
```

## 2. Add code that will iterate over all styles

Below lines that starts with `import` or `from` paste the following:
```
def queue_start(*args):
    for style in legal_style_names:
        argss = list(args)
        argss[2] = [style]
        yield from generate_clicked(*argss)
```

## 3. Override behavior of Generate button

Find the line starting with `generate_button.click(lambda` and a few lines below replace
`.then(fn=generate_clicked` 
with
`.then(fn=queue_start`

## 4. Run Fooocus with run.bat as normal

Enter any prompt you like and generate. For better comparison disable random seed. Every single style will be used once, no matter what you select in Style tab.

**Comment by mashb1t** [comment]
@docppp thank you for the explanation, much appreciated!
Quick relation link: https://github.com/lllyasviel/Fooocus/issues/1664#issuecomment-1873379637

**Comment by simartem** [comment]
@docppp i want to thank you for your effort and code works great! I just made the required changes and when i hit generate, it generates all styles. The main problem for me is, i have to revert to backup version of the "webui.py" to be using the version (without generating all the styles). Is it possible for you to add a checbox enabled function to generate all the styles, instead of modifying the generate button function ? So that, when the checbox is enabled --> " .then(fn=queue_start " activates. I dont know if its possible. Because this way, i will not have to revert between the backup webui and the modified one. I want to thank you again for your efforts.

...or instead of putting a "checkbox enabled", may be an additional button as "Generate-ALL" button, for the function

` .then(fn=queue_start `
 
to proceed..

**Comment by xhoxye** [comment]
When you click Stop, it can't stop, it can only close the program. And wouldn't it be nice to have a check box to tell if it works?

**Comment by docppp** [comment]
@simartem 
To rollback to the original behavior, all you need is to restore `.then(fn=generate_clicked`. The rest may stay as is. Please note that this is not (or even should imo) part of the source code from the main branch. And answering @xhoxye as well - no, this functionality cant be stopped in this form.

**Comment by andreafebrian** [comment]
can you please help to make it more clear? i have trouble to understand the step above. thank you

**Comment by simartem** [comment]
the main code in the first post used to work for me but somehow i updated fooocus and i cant run to generate "all styles" with one click. Is there any way to fix the code for latest version ?

**Comment by igor469** [comment]
Same problem. Doesn't work in version 2.3.1
It was very helpful.

**Comment by FrancK1207** [comment]
I wish to use it a picture that I have alreay made how can I do it? how can I restore the original settings?

**Reply by mashb1t** [reply]
This has nothing to do with this PR.

You can import the generation data using the metadata tab if you had enabled metadata generation when creating the image. If not you may be able to find the parameters in your history log.
